packages/hardhat-core/src/internal/sentry/anonymizer.ts

```diff
@@ -226,4 +234,80 @@ export class Anonymizer {
 
     return result;
   }
+
+  private _anonymizeMnemonic(
+    errorMessage: string,
+    mnemonicPhraseLengthThreshold: number,
+    minimumAmountOfWordsToAnonymize: number
+  ): string {
+    const fragments = errorMessage.split(/[!$%^&*()_+|~\-=`{}[\]:;<>?"',.]/gu);
+    const fragmentsToBeAnonymized: string[] = [];
+    // If there are 10 consecutive words, there's a good chance of there being a mnemonic phrase
+    const wordlist = ([] as string[]).concat(
+      ...[
+        require("ethereum-cryptography/bip39/wordlists/czech"),
+        require("ethereum-cryptography/bip39/wordlists/english"),
+        require("ethereum-cryptography/bip39/wordlists/french"),
+        require("ethereum-cryptography/bip39/wordlists/italian"),
+        require("ethereum-cryptography/bip39/wordlists/japanese"),
+        require("ethereum-cryptography/bip39/wordlists/korean"),
+        require("ethereum-cryptography/bip39/wordlists/simplified-chinese"),
+        require("ethereum-cryptography/bip39/wordlists/spanish"),
+        require("ethereum-cryptography/bip39/wordlists/traditional-chinese"),
+      ].map((wordlistModule) => wordlistModule.wordlist)
+    );
+    for (const [fragmentIndex, fragment] of fragments.entries()) {
+      // TODO: split on whitespace instead of spaces?
+      const words = fragment.split(" ");
+      if (words.length >= mnemonicPhraseLengthThreshold) {
+        // Determine all mnemonic phrase maximal fragments.
+        // We check sequences of n consecutive words just in case there is a typo
+        for (
+          let wordIndex = 0;
+          wordIndex < words.length - minimumAmountOfWordsToAnonymize + 1;
```

Comment:
I understand why the `- minimum...` is here, but it seems like a negligible optimization that only adds complexity. I'm ok with keeping it, but I would remove it.